### PR TITLE
NAS-102257

### PR DESCRIPTION
**Pull request by aervin**
* changes cloudsync defaults to PULL/COPY
* adds helptext paragraphs below transfer mode dropdown
* when user interacts with Direction dropdown, change transfer mode to COPY
* adds Transfer Mode column to cloudsync list (hidden by default)

### CODE REVIEW

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts [hunk 1]
```diff
@@ -67,6 +67,16 @@ transfer_mode_tooltip: T('<i>SYNC</i> makes files on the destination system iden
 
 transfer_mode_validation : [ Validators.required ],
 
+transfer_mode_warning_sync: T('<b>SYNC</b>: Files on the destination are <i>changed</i> to match those on the\
+ source. If a file does not exist on the source, it is also <i>deleted</i>\
```
Comment (a-stjohn) [inline src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]: Avoiding the "if-then" sentence.

```suggestion
 A file that does not exist on the source is <i>deleted</i> from\
```

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts [hunk 2]
```diff
@@ -67,6 +67,16 @@ transfer_mode_tooltip: T('<i>SYNC</i> makes files on the destination system iden
...
+ from the destination.'),
```
Comment (a-stjohn) [inline src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]: Avoiding the "if-then" sentence.
```suggestion
 the destination.'),
```

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts [hunk 3]
```diff
@@ -67,6 +67,16 @@ transfer_mode_tooltip: T('<i>SYNC</i> makes files on the destination system iden
...
+
+transfer_mode_warning_copy: T('<b>COPY</b>: Files from the source are <i>copied</i> to the destination. If files with\
```
Comment (a-stjohn) [inline src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]:
```suggestion
transfer_mode_warning_copy: T('<b>COPY</b>: Files from the source are <i>copied</i> to the destination. Files with\
```

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts [hunk 4]
```diff
@@ -67,6 +67,16 @@ transfer_mode_tooltip: T('<i>SYNC</i> makes files on the destination system iden
...
+ the same names are present on the destination, they are <i>overwritten</i>.'),
```
Comment (a-stjohn) [inline src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]:
```suggestion
 the same name on the destination are <i>overwritten</i>.'),
```

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts [hunk 5]
```diff
@@ -67,6 +67,16 @@ transfer_mode_tooltip: T('<i>SYNC</i> makes files on the destination system iden
...
+
+transfer_mode_warning_move: T('<b>MOVE</b>: After files are <i>copied</i> from the source to the destination, they are\
```
Comment (a-stjohn) [inline src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]:
```suggestion
transfer_mode_warning_move: T('<b>MOVE</b>: Files are <i>moved</i> from the source to the destination. The files\
```

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts [hunk 6]
```diff
@@ -67,6 +67,16 @@ transfer_mode_tooltip: T('<i>SYNC</i> makes files on the destination system iden
...
+ <i>deleted</i> from the source. Files with the same names on the destination are <i>overwritten</i>.'),
```
Comment (a-stjohn) [inline src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]:
```suggestion
 moved are <i>deleted</i> from the source. Files with the same names on the destination are <i>overwritten</i>.'),
```

**Comment by aervin** [comment]
@freenas/docs-team whenever the doc updates are ready, feel free to commit them directly.

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts (hunk 1 above)
Comment (wblock) [reply src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]: Unfortunately, this is less clear.

File: src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts (hunk 2 above)
Comment (wblock) [reply src/app/helptext/task-calendar/cloudsync/cloudsync-form.ts]: As above...

**Comment by wblock** [comment]
It occurs to me that transfer_mode_tooltip is just a concatenation of the three mode warning strings with spaces between them.  I was going to just make it do that in code, but I'm not sure how that would affect the translation.  Maybe not a problem, and less repetition of text.

**Pull request merged**